pingcap/tidb
# ddl: fix admin repair table will reload fail on the other node

### What problem does this PR solve?

Problem Summary:
Now in TiDB cluster, if config one node as repair-mode node, the other node  could not reload the new schema info after the table has been repaired.

The reason is because
repair-node: information schema has filtered the repaired tables, so for itself, apply the create table logic is ok.
other-node:  information schema has included the repaired tables, so for them, should apply the drop table first, then create new table.

Root cause:
sortedTablesBuckets will append the new repaired table into it, when find table with `TableByID`, then binary search will get the first old one rather the new one with same table ID (so should drop it first)
 

### What is changed and how it works?

How it Works:  drop old table first when apply repaired table in `applydiff`

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)
```
1: tiup playgroumd --db 2 
2: kill the tidb node & subscribe the bin and start it again
3: restart one node with repair-mode
4: conduct the repair SQL on it
5: check whether the node could reload the repaired table 
6: check whether the other node could reload the repaired table 
```

### Release note 
- ddl: fix admin repair table will reload fail on the other node
